### PR TITLE
[WIP] Activate deactivate

### PR DESCRIPTION
**Pull request by iocanel**
It still needs testing, thus the WIP label.

### CODE REVIEW

**Comment by iocanel** [comment]
retest this

**Comment by iocanel** [comment]
retest this

**Comment by iocanel** [comment]
retest this

**Comment by iocanel** [comment]
retest this

**Comment by iocanel** [comment]
retest this

File: controllers/src/main/java/com/redhat/ipaas/controllers/DeleteHandler.java [hunk 1]
```diff
@@ -15,26 +15,55 @@
...
     }
 
     @Override
-    public Optional<Integration.Status> execute(Integration model) throws Exception {
-        return model.getDesiredStatus();
+    public Optional<Integration.Status> execute(Integration integration) throws Exception {
+        Integration.Status currentStatus = !openShiftService.deploymentConfigExists(integration.getName())
+            || openShiftService.deleteResources(integration.getName())
+            ? Integration.Status.Deleted
+            : Integration.Status.Pending;
```
Comment (kahboom) [inline controllers/src/main/java/com/redhat/ipaas/controllers/DeleteHandler.java]: Oh wow!

File: model/src/main/java/com/redhat/ipaas/model/integration/Integration.java [hunk 2]
```diff
@@ -33,22 +33,8 @@
 @JsonDeserialize(builder = Integration.Builder.class)
 public interface Integration extends WithId<Integration>, WithName, Serializable {
 
-    public static enum Status { Draft, Activated, Deactivated, Deleted};
+    enum Status { Draft, Pending, Activated, Deactivated, Deleted};
```
Comment (kahboom) [inline model/src/main/java/com/redhat/ipaas/model/integration/Integration.java]: Say what! Why hasn't this been reviewed and approved yet? 😱

**Comment by chirino** [comment]
retest this please

**Comment by iocanel** [comment]
retest this please

**Pull request merged**